### DISCUSSION TITLE
Directly into the bash without "exit" in MA5 environment.

### DISCUSSION THREAD

**Discussion by shangliang** [MadAnalysis/madanalysis5#221] [General]
Hi Jack, I have a question about MA5. When one uses the expert model of MA5 as following,

./bin/ma5 -Rf madanalysis5_hadron_card.dat

Then one will stay in the environment of "MG5_aMC>...  ". I am wondering if we can get into the bash terminal directly without inputting "exit" in "MG5_aMC > " environment.

Best,
Liang

**Comment by jackaraz** [comment]
Hi @shangliang, you are not using expert mode in the code snippet you just posted and ``MG5_aMC`` is madgraph prompter, not madanalysis they are not the same thing, I believe you are confusing the tools. Please refer to the manual.

**Reply by shangliang** [reply]
Hi Jack, I am so sorry for that I had a mistake. I am using the recasting mode of MA5 not expert mode. And the prompter is "ma5>" not "MG5_aMC". Best, Liang.

**Comment by shangliang** [comment]
Hi Jack, the content in madanalysis5_hadron_card.dat is,

set main.recast = on
set main.recast.store_events = False
import tag_1_pythia8_events.hepmc.gz as signal
set signal.xsection = 0.7
set main.recast.card_path = recasting_card.dat
submit ANALYSIS_0
exit

And the output is,

...
MA5:    Checking SampleAnalyzer output...
MA5:    Preparing data for the reports ...
MA5:    Well done! Elapsed time = 14.36 seconds
MA5: ma5>exit
MA5: 
ma5>

Still in the 'ma5' environment.

Best,
Liang

**Reply by jackaraz** [reply]
have you tried ``quit``?

**Reply by shangliang** [reply]
Hi Jack, as you suggested, I tried "quit", but it gives the following response,

...
MA5:    Checking SampleAnalyzer output...
MA5:    Preparing data for the reports ...
MA5:    Well done! Elapsed time = 14.40 seconds
MA5: ma5>quit
MA5: 
ma5>

Best,
Liang

**Reply by jackaraz** [reply]
Hi @shangliang, I can not know what's happening in your machine if you just say, "I did this and that happened" I need further information about the configuration of your OS and software, I'm not in your machine. Can you please provide more information in addition to running ma5 in debug mode?

**Reply by shangliang** [reply]
Hi Jack, thank you for your reply. 

If I use the command without debug, 
../madanalysis5/bin/ma5 -Rf madanalysis5_hadron_card.dat
MA5 runs normally. 

But If I add the debug mode, 
../madanalysis5/bin/ma5 -Rfd madanalysis5_hadron_card.dat
it complained,

...
MA5-DEBUG (detect_python.ExtractInfo() in detect_python.py::L64):   Python executable used: /opt/homebrew/opt/python@3.11/bin/python3.11
MA5-DEBUG (detect_python.ExtractInfo() in detect_python.py::L68):   sys.executable:         /opt/homebrew/opt/python@3.11/bin/python3.11
MA5-ERROR: python compiler not found. Please install it before using MadAnalysis 5
MA5:      - Python                   [FAILURE]
MA5-ERROR: This package is a mandatory package: MadAnalysis 5 can not run without it.

So I think it is python version which causes this problem ?

Best,
Liang

**Reply by shangliang** [reply]
Hi Jack, I have installed python 3.10 and MA5 does not complain in debug mode, but it still could not exit. This is relevant message reported by MA5,

(with exit)
...
MA5:    Checking SampleAnalyzer output...
MA5-DEBUG (recast_configuration.CheckFile() in recast_configuration.py::L683): Check file "/Users/a233/shl/eVLQ/EasyScan_HEP-master/madanalysis5/PAD/ANALYSIS_0/Output/SAF/signal/CLs_output.dat"...
MA5:    Preparing data for the reports ...
MA5:    Well done! Elapsed time = 13.59 seconds
MA5: ma5>exit
MA5-DEBUG (interpreter_base.precmd() in interpreter_base.py::L166): ['exit']
MA5: 
ma5>

(with quit)
...
MA5:    Checking SampleAnalyzer output...
MA5-DEBUG (recast_configuration.CheckFile() in recast_configuration.py::L683): Check file "/Users/a233/shl/eVLQ/EasyScan_HEP-master/madanalysis5/PAD/ANALYSIS_0/Output/SAF/signal/CLs_output.dat"...
MA5:    Preparing data for the reports ...
MA5:    Well done! Elapsed time = 13.91 seconds
MA5: ma5>quit
MA5-DEBUG (interpreter_base.precmd() in interpreter_base.py::L166): ['quit']
MA5: 
ma5>

Best,
Liang
